addons/stock/models/stock_picking.py

```diff
@@ -1173,7 +1173,16 @@ def button_validate(self):
 
     def _pre_action_done_hook(self):
         for picking in self:
-            if all(not move.picked for move in picking.move_ids):
+            has_quantity = False
+            has_pick = False
+            for move in picking.move_ids:
+                if move.picked:
+                    has_pick = True
+                if move.quantity:
+                    has_quantity = True
+                if has_quantity and has_pick:
+                    break
+            if has_quantity and not has_pick:
```

Comment:
wouldn't this work as well?
```suggestion
            if any(move.quantity > 0 for move in picking.move_ids):
```